### PR TITLE
feat: replace setup.py with pyproject.toml

### PR DESCRIPTION
**Pull request by joaoandre-avaiga**

### CODE REVIEW

**Review by toan-quach — DISMISSED** [review]
lgtm :D

File: taipy/gui/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+Source = "https://github.com/Avaiga/taipy"
+Download = "https://pypi.org/project/taipy/#files"
+Tracker = "https://github.com/Avaiga/taipy/issues"
+Security = "https://github.com/Avaiga/taipy?tab=security-ov-file#readme"
+"Release notes" = "https://docs.taipy.io/en/release-0.0.0/relnotes/"  # will be set dynamically
+
+[tool.setuptools]
+zip-safe = false
+
+[tool.setuptools.package-data]
```
Comment (FabienLelaquais) [inline taipy/gui/pyproject.toml:53]: I don't see the 'taipy/gui/viselements.json' file that is mandatory.
Maybe somewhere else?

File: pyproject.toml [hunk 2]
```diff
@@ -1,3 +1,67 @@
...
+Source = "https://github.com/Avaiga/taipy"
+Download = "https://pypi.org/project/taipy/#files"
+Tracker = "https://github.com/Avaiga/taipy/issues"
+Security = "https://github.com/Avaiga/taipy?tab=security-ov-file#readme"
+"Release notes" = "https://docs.taipy.io/en/release-0.0.0/relnotes/"  # version will be dynamically set
+
+[tool.setuptools.packages.find]
+include = ["taipy", "taipy.*"]
+
+[tool.setuptools.package-data]
```
Comment (FabienLelaquais) [inline pyproject.toml:56]: I don't see the 'taipy/gui_core/viselements.json' file that is mandatory.
Maybe somewhere else?

**Review by FabienLelaquais — DISMISSED** [review]
That's really cool!
Make sure my two comments are addressed though.

File: pyproject.toml (hunk 2 above)
Comment (dinhlongviolin1) [reply pyproject.toml:56]: it should be included by the manifest.in file

**Review by dinhlongviolin1 — DISMISSED** [review]
LGTM

File: taipy/gui/pyproject.toml (hunk 1 above)
Comment (joaoandre-avaiga) [reply taipy/gui/pyproject.toml:53]: I just put the same files that were referenced in setup.py. Which was only the version.json.
```
    data_files=[("version", ["version.json"])],
```
We do however have `viselements.json` from gui and gui_core on the respective MANIFEST.in files for taipy-gui and taipy.

**Pull request merged**